### PR TITLE
[SYCL][libdevice] Test libdevice jit-link mode.

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

We added option f[no-]sycl-device-lib-jit-link to control SYCL device library link mode. Currently, jit-link is disabled by default and we explicitly test this option in this PR.

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @againull 
Could you help review this patch?

Thanks very much.

File: SYCL/DeviceLib/cmath_test.cpp [hunk 1]
```diff
@@ -3,6 +3,11 @@
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
 
+// RUN: %clangxx -fsycl -fno-builtin -fsycl-device-lib-jit-link %s -o %t.out
```
Comment (againull) [inline SYCL/DeviceLib/cmath_test.cpp:6]: Could you please clarify why some of the tests have -fno-builtin flag and some of them don't? Is this a workaround for something?

Comment (jinge90) [reply SYCL/DeviceLib/cmath_test.cpp:6]: Hi, @againull 
To some math functions, clang may have different ways to handle.
1. convert it to a library function call
2. instruction expand, use several IR instructions to replace
The tests aim to test implementations in SYCL device library, we need to guarantee all functions to test are converted to corresponding library function call, so adding "-fno-builtin" to force to achieve it.

Thanks very much.

Comment (jinge90) [reply SYCL/DeviceLib/cmath_test.cpp:6]: Hi, @againull 
Ping~

**Review by againull — APPROVED** [review]

File: SYCL/DeviceLib/cmath_test.cpp (hunk 1 above)
Comment (againull) [reply SYCL/DeviceLib/cmath_test.cpp:6]: Thank you  for the explanation!

**Pull request merged**